src/future/batch_semaphore.rs

```diff
@@ -501,48 +555,9 @@ impl BatchSemaphore {
 
         match self.fairness {
             Fairness::StrictlyFair => {
-                // in a strictly fair mode we will always pick the first waiter
-                // in the queue, as long as there are enough permits available
-                while let Some(front) = state.waiters.front() {
-                    if front.num_permits <= state.permits_available.available() {
-                        let waiter = state.waiters.pop_front().unwrap();
-
-                        crate::annotations::record_semaphore_acquire_unblocked(
-                            state.id.unwrap(),
-                            waiter.task_id,
-                            waiter.num_permits,
-                        );
-
-                        // The clock we pass into the semaphore is the clock of the
-                        // waiter, corresponding to the point at which the waiter was
-                        // enqueued. The clock we get in return corresponds to the
-                        // join of the clocks of the acquired permits, used to update
-                        // the waiter's clock to causally depend on the release events.
-                        let clock = state
-                            .permits_available
-                            .acquire(waiter.num_permits, waiter.clock.clone())
-                            .unwrap();
-                        trace!("granted {:?} permits to waiter {:?}", waiter.num_permits, waiter);
-
-                        // Update waiter state as it is no longer in the queue
-                        assert!(waiter.is_queued.swap(false, Ordering::SeqCst));
-                        assert!(!waiter.has_permits.swap(true, Ordering::SeqCst));
-                        ExecutionState::with(|s| {
-                            let task = s.get_mut(waiter.task_id);
-                            assert!(!task.finished());
-                            // The acquiry is causally dependent on the event
-                            // which released the acquired permits.
-                            task.clock.update(&clock);
-                            task.unblock();
-                        });
-                        let mut maybe_waker = waiter.waker.lock().unwrap();
-                        if let Some(waker) = maybe_waker.take() {
-                            waker.wake();
-                        }
-                    } else {
-                        break;
-                    }
-                }
+                // in a strictly fair mode we will grant permits to waiters from the front
+                // of the queue, as long as there are enough permits available
```

Comment:
This comment is now repeated here and in `unblock_waiters_from_front`. We probably want it removed in `unblock_waiters_from_front` ?

Reply:
Oops - fixed.